whelk-core/src/main/groovy/whelk/component/PostgreSQLComponent.groovy

```diff
@@ -55,6 +59,9 @@ class PostgreSQLComponent implements Storage {
     private static final String driverClass = "org.postgresql.Driver"
 
     private HikariDataSource connectionPool
+    private HikariDataSource outerConnectionPool
+    private final Object outerConnectionPoolLock = new Object()
```

Comment:
This is only used when creating a new outer pool, no? There's probably no need for a separate locking-object then, "this" could be enough.

Reply:
True, YAGNI